cscs-checks/tools/profiling_and_debugging/gperftools.py

```diff
@@ -0,0 +1,71 @@
+import os
+
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+
+@rfm.required_version('>=2.14')
+@rfm.parameterized_test(*([lang] for lang in ['C', 'Cpp', 'F90']))
+class GperftoolsCheck(rfm.RegressionTest):
+    def __init__(self, lang):
+        super().__init__()
+        self.valid_systems = ['daint:gpu', 'dom:gpu',
+                              'daint:mc', 'dom:mc']
+
+        self.valid_prog_environs = ['PrgEnv-cray', 'PrgEnv-gnu',
+                                    'PrgEnv-intel', 'PrgEnv-pgi']
+
+        # NOTE: Reduce time limit because for PrgEnv-pgi even if the output
```

Comment:
The time limit hack is not needed for this test. It runs fine without it.

Reply:
https://github.com/eth-cscs/reframe/pull/670/commits/70bb7ad6c774c66f945b0ef128f8a1562a9c7339